### DISCUSSION TITLE
OperatingSystemVersion is ?? not 10.0

### DISCUSSION THREAD

**Discussion by AndrzejKarwoski** [nielsbasjes/yauaa#823] [Q&A] [answered]
Hi,

I am trying to implement your library to my project but when I trying to pare User-Agent header i have bad output for Operating System Values

User-Agent                           : 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/107.0.0.0 Safari/537.36'
    expected:
      DeviceClass                          : 'Desktop'
      DeviceName                           : 'Desktop'
      DeviceBrand                          : 'Unknown'
      DeviceCpu                            : 'Intel x86_64'
      DeviceCpuBits                        : '64'
      OperatingSystemClass                 : 'Desktop'
      OperatingSystemName                  : 'Windows NT'
      OperatingSystemVersion               : '??'
      OperatingSystemVersionMajor          : '??'
      OperatingSystemNameVersion           : 'Windows NT ??'
      OperatingSystemNameVersionMajor      : 'Windows NT ??'
      OperatingSystemVersionBuild          : '??'

When i remove NT from header then all params are recognized correctly.

Is there is a bug, or I am doing somtehing wrong?

**Comment by nielsbasjes** [comment, marked as answer]
Have you read this?
https://yauaa.basjes.nl/developer/reportingissues/#these-are-not-bugs
